pgml-dashboard/src/lib.rs

```diff
@@ -119,6 +118,44 @@ impl Clusters {
     }
 }
 
+// Globally shared state in memory.
+
+#[derive(Debug, Clone)]
+pub struct CurrentUser {
```

Comment:
I'm not sure what this does. Is this meant to store the current user in the HTTP request?

Reply:
It is a global state to hold the currently signed in user and the clusters they have visible to them.  It was inside cluster context before, but you can have a user without a chosen cluster.

Reply:
I understand now. This is not going to work as is, I think. This global is global for everyone using the dashboard. In our cloud, this can be many different people. In this case, person A can land on the dashboard and we set their user as `test@test.com`. Then person B lands on the dashboard and we set this global user, in the same memory space, to `test2@test.com` and this overrides the same setting for person A.

What we do in the `Clusters` context is we set the current cluster in a `HashMap` with the `cluster_id` <-> `Cluster` object mapping. So when a request comes in, we fetch the request-specific object from the HashMap using the `cluster_id` key.

The global in that case stores _everyone_'s data and we retrieve the data specific to the user from that global.

Reply:
ok I see. I think a lot of this code is still usable.  I could probably change CurrentUser to take a HashMap, more like Clusters, and adjust the getter and setter methods.